sigar_aix.go

```diff
@@ -0,0 +1,538 @@
+// +build aix
+
+package gosigar
+
+/*
+#cgo LDFLAGS: -L/usr/lib -lperfstat
+
+#include <libperfstat.h>
+#include <procinfo.h>
+#include <unistd.h>
+#include <utmp.h>
+#include <sys/mntctl.h>
+#include <sys/proc.h>
+#include <sys/types.h>
+#include <sys/vmount.h>
+
+*/
+import "C"
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os"
+	"os/user"
+	"runtime"
+	"strconv"
+	"syscall"
+	"time"
+	"unsafe"
+)
+
+var system struct {
+	ticks    uint64
+	btime    uint64
+	pagesize uint64
+}
+
+func init() {
+	system.ticks = uint64(C.sysconf(C._SC_CLK_TCK))
+	system.pagesize = uint64(os.Getpagesize())
+	system.btime = bootTime()
+}
+
+// utmp can't be used by "encoding/binary" if generated by cgo,
+// some pads will be explicitly missing.
+type utmp struct {
+	User            [256]uint8
+	Id              [14]uint8
+	Line            [64]uint8
+	X_Pad1          int16
+	Pid             int32
+	Type            int16
+	X_Pad2          int16
+	Time            int64
+	Termination     int16
+	Exit            int16
+	Host            [256]uint8
+	X__dbl_word_pad int32
+	X__reservedA    [2]int32
+	X__reservedV    [6]int32
+}
+
+func bootTime() uint64 {
+	// Get boot time from /etc/utmp
+	file, err := os.Open("/etc/utmp")
+	if err != nil {
+		return 0
+	}
+
+	defer file.Close()
+
+	for {
+		var utmp utmp
+		if err := binary.Read(file, binary.BigEndian, &utmp); err != nil {
+			break
+		}
+
+		if utmp.Type == C.BOOT_TIME {
+			return uint64(utmp.Time)
+		}
+	}
+	return 0
+}
+
+func tick2msec(val uint64) uint64 {
```

Comment:
Could you please provide reference for this value as a comment?

Reply:
I've actually explained that sysconf(SC_CLK_TCK) is returning ticks/second in init().